core/src/main/java/io/questdb/griffin/OperatorRegistry.java

```diff
@@ -0,0 +1,110 @@
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+package io.questdb.griffin;
+
+import io.questdb.std.LowerCaseAsciiCharSequenceObjHashMap;
+import io.questdb.std.ObjList;
+
+public class OperatorRegistry {
+    public final OperatorExpression unaryMinus;
+    public final OperatorExpression unaryComplement;
+    public final OperatorExpression setOperationNegation;
+    public final OperatorExpression dot;
+    public final ObjList<OperatorExpression> operators;
+    public final LowerCaseAsciiCharSequenceObjHashMap<OperatorExpression> map;
+
+    public OperatorRegistry(ObjList<OperatorExpression> operators) {
+        this.map = new LowerCaseAsciiCharSequenceObjHashMap<OperatorExpression>();
+        this.operators = new ObjList<OperatorExpression>();
+        OperatorExpression dot = null, unaryMinus = null, unaryComplement = null, unarySetNegation = null;
+        for (int i = 0, k = operators.size(); i < k; i++) {
+            OperatorExpression op = operators.getQuick(i);
+            switch (op.operator) {
+                case UnaryMinus:
+                    unaryMinus = op;
+                    continue;
+                case UnaryComplement:
+                    unaryComplement = op;
+                    continue;
+                case UnarySetNegation:
+                    unarySetNegation = op;
+                    continue;
+            }
+            assert !map.contains(op.operator.token) : "unexpected operator conflict";
+            map.put(op.operator.token, op);
+            this.operators.add(op);
+            if (op.operator == OperatorExpression.Operator.Dot) {
+                dot = op;
+            }
+        }
+
+        assert dot != null : "dot operator ('.') must be present in operators list";
+        assert unaryMinus != null : "unary minus operator ('-') must be present in operators list";
+        assert unaryComplement != null : "unary complement operator ('~') must be present in operators list";
+        assert unarySetNegation != null : "unary set negation operator (e.g. 'not within') must be present in operators list";
+
+        this.dot = dot;
+        this.unaryMinus = unaryMinus;
+        this.unaryComplement = unaryComplement;
+        this.setOperationNegation = unarySetNegation;
+    }
+
+    public int getOperatorType(CharSequence name) {
+        int index = map.keyIndex(name);
+        if (index < 0) {
+            return map.valueAt(index).type;
+        }
+        return 0;
+    }
+
+    public OperatorExpression tryGetOperator(OperatorExpression.Operator operator) {
```

Comment:
This method, as well as `tryGuessOperator`, deserve a javadoc clarifying their purpose.